### PR TITLE
NVD Scanning

### PR DESCRIPTION
**Pull request by milt**
Investigate generalizing `nvd-clojure` scans on our OSS clj projects with gh actions.

See scheduled runs here: https://github.com/yetanalytics/xapipe/actions/workflows/nvd_sched.yml

### CODE REVIEW

**Review by cliffcaseyyet — APPROVED** [review]
word

File: .github/workflows/nvd-scan.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+      - name: Run NVD scan
+        run: clojure -Sdeps '{:aliases {:nvd {:replace-deps {nvd-clojure/nvd-clojure {:mvn/version "${{ inputs.nvd-clojure-version }}"}}}}}' -Xnvd nvd.task/check :classpath '"'"$(${{ inputs.classpath-command }})"'"'
+
+      - name: Produce Artifact
+        uses: actions/upload-artifact@v2
+        if: ${{ always() }}
+        with:
+          name: ${{ github.sha }}-nvd-report
+          path: target/nvd
```
Comment (kelvinqian00) [inline .github/workflows/nvd-scan.yml]: Huh curious about this about why this `if true` conditional exists.

File: .github/workflows/nvd-scan.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            ~/.m2
+            ~/.gitlibs
+          key: ${{ runner.os }}-nvd-all-deps-${{ steps.date.outputs.y }}-${{ steps.date.outputs.m }}-${{ steps.date.outputs.d }}
+          restore-keys: |
+            ${{ runner.os }}-nvd-all-deps-${{ steps.date.outputs.y }}-${{ steps.date.outputs.m }}-
+            ${{ runner.os }}-nvd-all-deps-${{ steps.date.outputs.y }}-
+            ${{ runner.os }}-nvd-all-deps-
+
+      - name: Run NVD scan
+        run: clojure -Sdeps '{:aliases {:nvd {:replace-deps {nvd-clojure/nvd-clojure {:mvn/version "${{ inputs.nvd-clojure-version }}"}}}}}' -Xnvd nvd.task/check :classpath '"'"$(${{ inputs.classpath-command }})"'"'
```
Comment (kelvinqian00) [inline .github/workflows/nvd-scan.yml]: Might want to separate into different lines using `run: >`

File: .github/workflows/nvd-scan.yml (hunk 1 above)
Comment (milt) [reply .github/workflows/nvd-scan.yml]: This is how you tell it to produce an artifact, even on failure. I agree that it isn't clear, so I'll add a comment about it.

File: .github/workflows/nvd-scan.yml (hunk 2 above)
Comment (milt) [reply .github/workflows/nvd-scan.yml]: I keep forgetting all the sugar YAML gives you. Sure I'll rot my teeth.

Comment (milt) [reply .github/workflows/nvd-scan.yml]: hmm, I did that but now: https://github.com/yetanalytics/xapipe/runs/4652213511?check_suite_focus=true

Comment (milt) [reply .github/workflows/nvd-scan.yml]: One thing I think this suggests is that I should set up CI in this project for the actions themselves, using a dummy clojure proj that follows our conventions

Comment (milt) [reply .github/workflows/nvd-scan.yml]: https://github.com/yetanalytics/actions/pull/2/commits/c920fd070e052f21f70a853e8d79b25c25f45df6

File: .github/workflows/nvd-scan.yml (hunk 1 above)
Comment (milt) [reply .github/workflows/nvd-scan.yml]: https://github.com/yetanalytics/actions/pull/2/commits/e34771b9e840e5506809ea7c8929981ff33e0db8

File: .github/workflows/nvd-scan.yml (hunk 2 above)
Comment (kelvinqian00) [reply .github/workflows/nvd-scan.yml]: Interesting that removing the indentation makes it work. YAML is weird...

**Review by kelvinqian00 — APPROVED** [review]

File: .github/workflows/nvd-scan.yml (hunk 2 above)
Comment (milt) [reply .github/workflows/nvd-scan.yml]: Yeah it looks like `>` makes a newline on any change in indent

**Pull request merged**